Templates for IOMAD (single VM and cluster)

vinhub opened the pull request.

azurecla commented:
Hi __@vinhub__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

singhkay commented on iomad-cluster-ubuntu/azuredeploy.json:
```diff
@@ -0,0 +1,673 @@
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "newStorageAccountName": {
```
Please remove. See #944

singhkay commented on iomad-cluster-ubuntu/azuredeploy.json:
```diff
@@ -0,0 +1,673 @@
+    "moodleAdminUsername":{
+      "type": "string",
+      "metadata": {
+        "description": "User name for the IOMAD site administrator."
```
parameter is moodle but description is IOMAD. Is that expected?

singhkay commented on iomad-singlevm-ubuntu/azuredeploy.json:
```diff
@@ -0,0 +1,243 @@
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "newStorageAccountName": {
```
Same comments as above template

vinhub commented:
Converted storage account name into variable
Renamed moodle to iomad where it made sense. Iomad is the multi tenant version of moodle so we had kept the name, but probably a good idea to name it as iomad to avoid confusion.

singhkay commented:
@sedouard Finally caught a Server Error:undefined error! Can you take a look?

singhkay commented:
@vinhub When the CI tries to deploy, it's getting this error. There's some file in the custom script that it is not able to download. Can you check?

```
"Enable failed: Failed with error: HTTP Error 404: Not Found, Traceback (most recent call last):\n  File \"/var/lib/waagent/Microsoft.OSTCExtensions.CustomScriptForLinux-1.4.1.0/customscript.py\", line 83, in main\n    daemon(hutil)\n  File \"/var/lib/waagent/Microsoft.OSTCExtensions.CustomScriptForLinux-1.4.1.0/customscript.py\", line 284, in daemon\n    retry_count = download_files_with_retry(hutil, retry_count, wait)\n  File \"/var/lib/waagent/Microsoft.OSTCExtensions.CustomScriptForLinux-1.4.1.0/customscript.py\", line 123, in download_files_with_retry\n    download_files(hutil)\n  File \"/var/lib/waagent/Microsoft.OSTCExtensions.CustomScriptForLinux-1.4.1.0/customscript.py\", line 229, in download_files\n    download_external_files(blob_uris, cmd, hutil)\n  File \"/var/lib/waagent/Microsoft.OSTCExtensions.CustomScriptForLinux-1.4.1.0/customscript.py\", line 430, in download_external_files\n    download_external_file(uri, command, hutil)\n  File \"/var/lib/waagent/Microsoft.OSTCExtensions.CustomScriptForLinux-1.4.1.0/customscript.py\", line 440, in download_external_file\n    download_and_save_file(uri, file_path)\n  File \"/var/lib/waagent/Microsoft.OSTCExtensions.CustomScriptForLinux-1.4.1.0/customscript.py\", line 451, in download_and_save_file\n    src = urllib2.urlopen(uri, timeout=timeout)\n  File \"/usr/lib/python2.7/urllib2.py\", line 127, in urlopen\n    return _opener.open(url, data, timeout)\n  File \"/usr/lib/python2.7/urllib2.py\", line 410, in open\n    response = meth(req, response)\n  File \"/usr/lib/python2.7/urllib2.py\", line 523, in http_response\n    'http', request, response, code, msg, hdrs)\n  File \"/usr/lib/python2.7/urllib2.py\", line 448, in error\n    return self._call_chain(*args)\n  File \"/usr/lib/python2.7/urllib2.py\", line 382, in _call_chain\n    result = func(*args)\n  File \"/usr/lib/python2.7/urllib2.py\", line 531, in http_error_default\n    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)\nHTTPError: HTTP Error 404: Not Found\n"
```

singhkay commented:
lgtm, merged
